### PR TITLE
Add allowed users example to Users resource

### PR DESCRIPTION
**Pull request by shoekstra**
This example shows how to check for expected users and to flag any other
users that should not be present.

Signed-off-by: Stephen Hoekstra <stephenhoekstra@gmail.com>

### CODE REVIEW

**Comment by jerryaldrichiii** [comment]
Brining the conversation out of Slack into here 😄 

I'm not sure I like the `if/else` bits...but it's not wrong per se.

Here's how I would have done it:

```ruby
### Ensure only users in allowed_users exist (with UIDs > 500)

    allowed_users = %w{jerry boiarde}
    current_users = users.where { uid >= 1000 }.usernames
    invalid_users = users.where { uid >= 1000 && !allowed_users.include?(username) }.usernames

    describe 'Users with uid >= 500' do
      it 'should only contain allowed users' do
        expect(invalid_users).to be_empty
      end
      it 'should contain all allowed users' do
        expect(current_users).to include(allowed_users)
      end
    end
```

Thoughts?

EDIT:

Output from command (I'm still not happy with it):
```
  Users with uid >= 500
     ×  should only contain allowed users
     expected `["nobody"].empty?` to return true, got false
     ×  should contain all allowed users
     expected ["nobody"] to include ["jerry", "boiarde"]
```

**Review by miah — APPROVED** [review]
Thanks @shoekstra

File: docs/resources/users.md.erb [hunk 1]
```diff
@@ -58,6 +58,20 @@ The following examples show how to use this Chef InSpec audit resource.
       it { should exist }
     end
 
+### Test only allowed users exist
+
+    allowed_users = %w(user1 user2 user3)
+
+    users.where { uid > 1000 && uid < 65534 }.usernames.sort.each do |u|
```
Comment (jerryaldrichiii) [inline docs/resources/users.md.erb:65]: No need for the upper bound I suspect.

**Review by jerryaldrichiii — APPROVED** [review]

**Comment by jerryaldrichiii** [comment]
I changed my mind, your example is simpler.

**Comment by aaronlippold** [comment]
Actually, wouldn't it be better to use the be_in matcher here? Provide much better reporting output.

File: docs/resources/users.md.erb (hunk 1 above)
Comment (shoekstra) [reply docs/resources/users.md.erb:65]: Maybe not, in our environment it's the `nfsnobody` user but happy to omit it for the example.

**Comment by shoekstra** [comment]
@aaronlippold when I tested locally I got this:

```
Profile: inspec-shell
Version: (not specified)

  User test1
     ✔  should exist
  User test2
     ✔  should exist
...
  User test3
     ×  should not exist
     expected User test3 not to exist

Test Summary: 18 successful, 2 failures, 0 skipped
```

I'm happy with this output.

@jerryaldrichiii I like what you did and it may have been something I would do, but for the example wanted to keep it short and to the point like other examples I've come across.

**Pull request merged**